rclcpp/include/rclcpp/node.hpp

```diff
@@ -1448,6 +1449,39 @@ class Node : public std::enable_shared_from_this<Node>
   rclcpp::node_interfaces::NodeTimeSourceInterface::SharedPtr
   get_node_time_source_interface();
 
+  /// Return a NodeHandle bound with the Node's internal node interfaces.
+  /**
+   * This overload binds all interfaces, if you want to bind a subset of interfaces, specify them
+   * in the template parameters of the templated method.
+   *
+   * \return a NodeHandle bound with the Node's internal node interfaces.
+   */
+  RCLCPP_PUBLIC
+  inline
+  rclcpp::NodeHandle<AllInterfaces>::SharedPtr
+  get_node_handle() {return std::make_shared<NodeHandle<AllInterfaces>>(this);}
+
+  /// Return a NodeHandle bound with the Node's internal node interfaces.
+  /**
+   * Specify which interfaces you want to bind using the temlplate parameters. Any unbound
+   * interfaces will be nullptr.
+   *
+   * Additionally, to bind all interfaces, specify AllInterfaces or no template parameters.
+   *
+   * For example:
+   *   - ```node->get_node_handle<>()``` will bind no interfaces.
+   *   - ```node->get_node_handle<BaseInterface>()``` will bind just the NodeBaseInterface.
+   *   - ```node->get_node_handle<BaseInterface, ClockInterface>()``` will bind the base and clock
+   *     interfaces.
+   *   - ```node->get_node_handle<AllInterfaces>()``` will bind all interfaces.
+   *   - ```node->get_node_handle()``` will bind all interfaces.
+   *
+   * \return a NodeHandle bound with the Node's internal node interfaces.
+   */
+  template<typename ... InterfaceTs>
+  typename rclcpp::NodeHandle<InterfaceTs...>::SharedPtr
+  get_node_handle();
+
```

Comment:
It would be better (in my opinion) for the node class to know nothing about the new concept (`NodeHandle`). I would strive to remove functions like these. Instead a constructor for `NodeHandle` that can take any class that looks like `Node` or `LifecycleNode` would be better.

Reply:
Hmm, I was thinking along the lines of the NodeHandle class being like an aggregate interface you can get, just as with the node interface getters.

But since a constructor taking in NodeT on NodeHandle already exists, I'm fine with removing the ones local to each Node/LifecycleNode class. They're more for convenience anyway.

Reply:
I still think remove them.

I really think Node should not have to know about them, because every future Node-like class will also have to add them to keep up with LifecycleNode and Node. It just doesn't scale and it doesn't follow encapsulation.

Reply:
Will remove!